views/register.ejs

```diff
@@ -3,27 +3,38 @@
 
 <div class="container mt-5">
   <form method="POST" class="col-md-6 mx-auto">
+    <input type="hidden" name="_csrf" value="<%= csrfToken %>">
     <div class="form-group">
       <label for="name">Enter your name:</label>
-      <input type="text" class="form-control" id="name" name="name" placeholder="Enter your name">
+      <input type="text" class="form-control" id="name" name="name" placeholder="Enter your name" required>
     </div>
     <div class="form-group">
       <label for="email">Enter your email:</label>
-      <input type="email" class="form-control" id="email" name="email" placeholder="Enter your email">
+      <input type="email" class="form-control" id="email" name="email" placeholder="Enter your email" required>
     </div>
     <div class="form-group">
       <label for="password">Enter your password:</label>
-      <input type="password" class="form-control" id="password" name="password" placeholder="Enter your password">
+      <input type="password" class="form-control" id="password" name="password" placeholder="Enter your password" required>
     </div>
     <div class="form-group">
       <label for="password1">Confirm your password:</label>
-      <input type="password" class="form-control" id="password1" name="password1" placeholder="Confirm your password">
+      <input type="password" class="form-control" id="password1" name="password1" placeholder="Confirm your password" required>
     </div>
     <div class="form-group">
       <button type="submit" class="btn btn-primary">Register</button>
       <a href="/" class="btn btn-secondary">Cancel</a>
     </div>
+    <% if (errors && errors.length > 0) { %>
+      <div class="alert alert-danger">
```

Comment:
Nitpick:
This is an unnecessary `div` element. Instead, remove the div and apply the `class` attribute to the `ul` element.